uma/uma.go

```diff
@@ -1011,3 +1020,64 @@ func VerifyPostTransactionCallbackSignature(
 	}
 	return verifySignature(*signablePayload, *callback.Signature, otherVaspPubKeyResponse)
 }
+
+func CreateUmaInvoice(
+	receiverUma string,
+	amount uint64,
+	receivingCurrency protocol.InvoiceCurrency,
+	expiration uint64,
+	callback string,
+	isSubjectToTravelRule bool,
+	requiredPayerData *protocol.CounterPartyDataOptions,
+	commentCharsAllowed *int,
+	receiverKycStatus *protocol.KycStatus,
+	invoiceLimit *uint64,
+	senderUma *string,
+	signingPrivateKey []byte,
+) (string, error) {
+	uuid := uuid.New().String()
+	invoice := protocol.UmaInvoice{
+		ReceiverUma:           receiverUma,
+		InvoiceUUID:           uuid,
+		Amount:                amount,
+		ReceivingCurrency:     receivingCurrency,
+		Expiration:            expiration,
+		IsSubjectToTravelRule: isSubjectToTravelRule,
+		RequiredPayerData:     requiredPayerData,
+		// TODO: modify the version once ready, all current version cannot support UMA invoice.
+		// Since this only add fields and features to the protocol, this won't break the current
+		// UMA version so it can be a minor version bump.
+		UmaVersion:          UmaProtocolVersion,
+		CommentCharsAllowed: commentCharsAllowed,
+		SenderUma:           senderUma,
+		InvoiceLimit:        invoiceLimit,
+		KycStatus:           receiverKycStatus,
+		Callback:            callback,
+		Signature:           nil,
+	}
+	signablePayload, err := invoice.MarshalTLV()
+	if err != nil {
+		return "", err
+	}
+	signature, err := signPayloadToBytes(signablePayload, signingPrivateKey)
+	if err != nil {
+		return "", err
+	}
+	invoice.Signature = &signature
+	return invoice.ToBech32String()
+}
+
+func DecodeUmaInvoice(invoice string) (*protocol.UmaInvoice, error) {
+	return protocol.FromBech32String(invoice)
+}
+
+func VerifyUmaInvoiceSignature(invoice protocol.UmaInvoice, otherVaspPubKeyResponse protocol.PubKeyResponse) error {
```

Comment:
Should this be a function on the invoice object instead? Might make it easier for folks to find the method.

Reply:
I think for now I'm going to to the verification here as it creates another object to get the unsigned payload. Also this follows the patter for other signature validation in this file.